src/create-glamorous.js

```diff
@@ -7,11 +7,17 @@ import {PropTypes} from './react-compat'
 import withTheme from './with-theme'
 import getGlamorClassName from './get-glamor-classname'
 
+let glamorousInstance = 1
+
 export default createGlamorous
 
 function createGlamorous(splitProps) {
   // TODO: in a breaking version, make this default to true
-  glamorous.config = {useDisplayNameInClassName: false}
+  glamorous.config = {
+    useDisplayNameInClassName: false,
+    idCounter: 1,
+    instanceId: glamorousInstance++,
```

Comment:
The core issue is that, with server rendering, different requests would use the same ID sequence for these component IDs, while each client would start from scratch.

Two concrete examples where this can cause problems:

- You create styled components dynamically. With the same `glamorous` instance, on the server, they get increased IDs every time. On the client, they don't.
- You use code splitting. The server (eventually) loads up all the bundles, but the client only loads the bundles specific to it. Actually, this is even harder to deal with, because even if you could have the "correct" bundles loaded on the server, you'd need to ensure the client loads and evaluates the static bits in the same order.

Without digging too deeply into this code, I think it'd be relatively challenging to nicely rehydrate the component IDs. The core problem is that you don't have anything to uniquely identify the components – that's why you need that ID in the first place.

Part of the problem is that, with the current implementation, the IDs get generated statically (i.e. at module import time). You could imagine an approach where you did some sort of "replay" at the time you do the server render – for example, record the sequence of IDs that are generated, and send that to the client such that the client can "replay" that sequence of IDs, but at that point it's too late; you can't observe the order in which the components are statically generated.

If you had a different instance of glamorous for each server rendering request, then you'd avoid many of these problems (assuming you could deal with the "order of split-out bundles" if using code splitting), but I guess the API here doesn't naturally let you put this on context like people do with Flux. Zones/CLS would handle that nicely, except for how they don't really exist.